File: src/main/java/gui/NetworkPanel.java
package gui;

import network.Network;
import utils.PropertiesHandler;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;
import java.awt.*;

public class NetworkPanel extends JScrollPane {
    private final Network network;
    private final Render render;

    public NetworkPanel(@NotNull Network network) {
        super();
        this.network = network;
        render = new Render(network);
        viewport.setView(render);
        PropertiesHandler cfg = PropertiesHandler.getProps();
        setPreferredSize(new Dimension(cfg.getInt("viewW"),
                cfg.getInt("viewH")));
    }

    public void update() {
        render.update();
        revalidate();
    }

    public Network getNetwork() {
        return network;
    }
}

File: src/main/java/mvc/view/NetworkPanel.kt
package mvc.view

import mvc.controller.Network
import common.Config

import javax.swing.*
import java.awt.*

class NetworkPanel(val network: Network) : JScrollPane() {
    private val render: GraphImage = GraphImage(network)

    init {
        viewport.view = render
        val cfg = Config.config
        preferredSize = Dimension(cfg.getInt("viewW"),
                cfg.getInt("viewH"))
    }

    fun update() {
        render.update()
        revalidate()
    }
}
